apps/src/lab2/projects/ProjectManager.ts

```diff
@@ -574,11 +574,32 @@ export default class ProjectManager {
       // showing the user a dialog before reload.
       this.forceReloading = true;
       this.metricsReporter.logWarning(`${error.message}. Reloading page.`);
+      const errorType = error.message.includes('409')
+        ? 'conflict'
+        : 'unauthorized';
+      this.metricsReporter.publishMetric(
+        'Lab2.ProjectSaveFailureClient',
+        1,
+        'Count',
+        [
+          {name: 'SaveType', value: type},
+          {name: 'ErrorType', value: errorType},
+        ]
+      );
       reload();
     } else if (error.message.includes('413')) {
       // Log 413s as warnings. The save fail listener should handle these errors and labs should
       // show a reasonable error message to the user.
       this.metricsReporter.logWarning('Project too large to save');
+      this.metricsReporter.publishMetric(
```

Comment:
I'm not convinced we should log this one; we give a message to the user in this case and consider it a warning since it's on the user to decrease their project size. We also don't have a similar metric for legacy.

Reply:
Makes sense - removed!